crates/turborepo-auth/src/auth/mod.rs

```diff
@@ -5,3 +5,77 @@ mod sso;
 pub use login::*;
 pub use logout::*;
 pub use sso::*;
+use turborepo_api_client::Client;
+use turborepo_ui::{BOLD, UI};
+
+use crate::{ui, LoginServer};
+
+const VERCEL_TOKEN_DIR: &str = "com.vercel.cli";
+const VERCEL_TOKEN_FILE: &str = "auth.json";
+
+pub struct LoginOptions<'a, T>
+where
+    T: Client,
+{
+    pub ui: &'a UI,
+    pub login_url: &'a str,
+    pub api_client: &'a T,
+    pub login_server: &'a dyn LoginServer,
```

Comment:
I think I can do this "cleaner" in another PR, but I don't want to get too far out of the initial scope of the PR